### PR TITLE
Modernize Python 2 code to get ready for Python 3

### PR DESCRIPTION
**Pull request by cclauss**
Make the minimal, safe changes required to convert the repo's code to be syntax compatible with both Python 2 and Python 3. There would _definitely_ be much more work required to complete the port to Python 3 but this is a minimal, safe first step.

1. Run: `futurize --stage1 -w **/*.py`

See Stage 1: "safe" fixes http://python-future.org/automatic_conversion.html#stage-1-safe-fixes

$ `pip install future`
$ `git checkout -b modernize-python2-code`
$ `futurize --stage1 -w **/*.py` # done in zsh for ** globbing
$ `git commit --all -m "Modernize Python 2 code to get ready for Python 3"`
$ `git push --set-upstream origin modernize-python2-code`

2. Make the changes in #2389 to fix a few Python 3 syntax errors

### CODE REVIEW

**Comment by panyx0718** [comment]
LGTM

**Comment by lukaszkaiser** [comment]
LGTM on my side.

**Comment by cclauss** [comment]
@panyx0718 @lukaszkaiser I believe that you guys need to check the box in the upper right of this page

reviewers

File: neural_gpu/neural_gpu.py [hunk 1]
```diff
@@ -500,8 +500,10 @@ def attention_query(query, attn_v):
             return tf.reduce_sum(encoder_outputs * tf.expand_dims(mask, 2), 1)
 
           with tf.variable_scope("decoder"):
-            def decoder_loop_fn((state, prev_cell_out, _), (cell_inp, cur_tgt)):
+            def decoder_loop_fn(xxx_todo_changeme, xxx_todo_changeme1):
```
Comment (YuMS) [inline neural_gpu/neural_gpu.py]: It looks like your script left some todos here.

Comment (cclauss) [reply neural_gpu/neural_gpu.py]: These are temporary variables which are immediately consumed on the following line.  The names that I could think up for them (`state__prev_cell_out__unused` and `cell_imp__cur_tgt`) were not any more intuitive/ user friendly than the defaults.

Comment (YuMS) [reply neural_gpu/neural_gpu.py]: An alternative could be the names used for invoking `decoder_loop_fn` on line 543: 
`(embedded_dec_inp_proj, gpu_tgt_trans)`

See
https://github.com/Upgrade-to-Python3/models/blob/fbeb6ad6d13daa9708e32469db40a0cf52aadc6d/neural_gpu/neural_gpu.py#L543

ambitious organization name :)

**Comment by cclauss** [comment]
@OriolVinyals do these changes make sense to you?

**Comment by cclauss** [comment]
Hopelessly out of date :-(

**Pull request closed without merging**